pulser-simulation/pulser_simulation/simulation.py

```diff
@@ -100,7 +103,20 @@ def __init__(
         self._interaction = "XY" if self._seq._in_xy else "ising"
         self._qdict = self._seq.qubit_info
         self._size = len(self._qdict)
-        self._tot_duration = self._seq.get_duration()
+        self._modulated = bool(with_modulation)
```

Comment:
Since this is a public method, I figured I should at least try to account for the case where someone doesn't respect the type hint. Truth be told though, everything can be cast to a bool, so I can see it being the wrong approach here. Do you think I should explicitly check that it is a bool instead?

Reply:
Well by doing so then you then consider explicitly checking types of all arguments of public methods, in addition to the type hinting and the documentation.
This is python's peculiarity to have type hints unrelevant at runtime, they basically give no additional real guarantees, despite being quite useful to the users.

So I would say you should just assume that `with_modulation` is a boolean and add a type hint to `self._modulated` if it is inherited from `with_modulation`.

Reply:
Ok, agreed.